lib/pages/contacts_tab/contacts_tab.dart

```diff
@@ -71,6 +75,52 @@ class ContactsTabController extends State<ContactsTab>
     });
   }
 
+  void onContactTap({
+    required BuildContext context,
+    required String path,
+    required PresentationContact contact,
+  }) {
+    final roomId =
+        Matrix.of(context).client.getDirectChatFromUserId(contact.matrixId!);
+    if (roomId == null) {
+      goToDraftChat(
+        context: context,
+        path: path,
+        contact: contact,
+      );
+    } else {
+      showFutureLoadingDialog(
+        context: context,
+        future: () async {
+          if (contact.matrixId != null && contact.matrixId!.isNotEmpty) {
+            context.go('/$path/$roomId');
+          }
+        },
+      );
+    }
+  }
+
+  void goToDraftChat({
+    required BuildContext context,
+    required String path,
+    required PresentationContact contact,
+  }) {
+    if (contact.matrixId != Matrix.of(context).client.userID) {
+      Router.neglect(
```

Comment:
what do this do ?

Reply:
<img width="696" alt="Screenshot 2023-08-11 at 09 46 28" src="https://github.com/linagora/twake-on-matrix/assets/99852347/c4d8ba9d-3740-428c-8ed7-36355819e1dc">